### ISSUE TITLE
[Terminal Finder] ...

### ISSUE THREAD

**Issue by hugosaner** [raycast/extensions#2540]
# Extension – [Terminal Finder](https://github.com/raycast/extensions/tree/de047f2e9ad02d2c79a526f2291f64f74aca05b6/extensions/terminalfinder/)
Author:

## Description

On a MacBook Pro M1 with the finder window open using the terminal finder extension to Finder -> Terminal it pops up an error stating that the Finder window is not open and fails to open the terminal.

This worked perfectly on my intel MacBook Pro 

## Steps To Reproduce

1. Find the folder in finder that you want to open
2. Execute the Terminal Finder command to open in terminal3. 
3. See the error message

## The current behavior

See description

## The expected behavior

I expect it to open the terminal at the same directory location as show in the finder window

### Raycast version

1.38.1

**Comment by frypf** [comment]
I noticed this too on my M1 Air (although I was attempting to open the path in iTerm) if I have Finder's Inspector window open.

I copied the AppleScript command from source and ran it in Script Editor, and realised it was indeed trying and failing to grab a path from the Inspector window. Changing [this line](https://github.com/raycast/extensions/blob/de047f2e9ad02d2c79a526f2291f64f74aca05b6/extensions/terminalfinder/src/fi.tsx#L11) to `set pathList to (quoted form of POSIX path of (folder of the front Finder window as alias))` seems to work.

**Comment by hugosaner** [comment]
Thanks for the help frypf. Here's hoping the fix is added soon

**Comment by hugosaner** [comment]
Ah I found the problem that I was facing. I updated my computer to the M1 and just ported everything over. But by doing that all the permissions I granted for things like finder access for this extension and music permission didn't port over and so the extensions stopped working.

I fixed it by backing up my settings etc and then uninstalling and reinstalling all the extensions. Everything works fine now
